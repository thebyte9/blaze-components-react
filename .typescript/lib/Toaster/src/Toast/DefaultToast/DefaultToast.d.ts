/// <reference types="react" />
import { IToastProps } from "../types/common";
declare const DefaultToast: {
    ({ appearance, autoDismiss, autoDismissTimeout, children, isRunning, onDismiss, placement, transitionDuration, transitionState, onMouseEnter, onMouseLeave, ...otherProps }: IToastProps): JSX.Element;
    defaultProps: {
        onDismiss: () => void;
    };
};
export default DefaultToast;
