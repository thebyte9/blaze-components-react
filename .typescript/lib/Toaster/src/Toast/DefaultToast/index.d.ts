import DefaultToast from "./DefaultToast";
export default DefaultToast;
