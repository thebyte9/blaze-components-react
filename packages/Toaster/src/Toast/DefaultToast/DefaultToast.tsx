import React from "react";
import Icon from "../../Icon";
import { CloseIcon } from "../../Icons";
import ToastElement from "../ToastElement";
import { IToastProps } from "../types/common";
import { NOOP } from "../utils";

const DefaultToast = ({
  appearance,
  autoDismiss,
  autoDismissTimeout,
  children,
  isRunning,
  onDismiss,
  placement,
  transitionDuration,
  transitionState,
  onMouseEnter,
  onMouseLeave,
  ...otherProps
}: IToastProps) => (
  <ToastElement
    appearance={appearance}
    placement={placement}
    transitionState={transitionState}
    transitionDuration={transitionDuration}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
    {...otherProps}
  >
    <Icon
      appearance={appearance}
      autoDismiss={autoDismiss}
      autoDismissTimeout={autoDismissTimeout}
      isRunning={isRunning}
    />
    <div className="react-toast-notifications__toast__content">{children}</div>
    {onDismiss ? (
      <div
        role="button"
        className="react-toast-notifications__toast__dismiss-button"
        onClick={onDismiss}
      >
        <CloseIcon
          className={`react-toast-notifications__toast__dismiss-icon react-toast-notifications__toast__dismiss-icon--${appearance}`}
        />
        <span className="react-toast-notifications__toast__dismiss-text" />
      </div>
    ) : null}
  </ToastElement>
);

DefaultToast.defaultProps = {
  onDismiss: NOOP
};

export default DefaultToast;
