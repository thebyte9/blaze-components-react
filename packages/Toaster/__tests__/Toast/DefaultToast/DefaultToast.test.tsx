import { render } from "@testing-library/react";
import expect from "expect";
import React from "react";
import DefaultToast from "../../../src/Toast/DefaultToast";

describe("Default toast", () => {
  it("should be defined", () => {
    expect(DefaultToast).toBeDefined();
  });

  it("should render without throwing error", () => {
    const { container } = render(
      <DefaultToast
        appearance={"warning"}
        autoDismiss={true}
        autoDismissTimeout={5000}
        isRunning={true}
        onDismiss={jest.fn()}
        onMouseEnter={jest.fn()}
        onMouseLeave={jest.fn()}
        placement="top-right"
        transitionDuration={220}
        transitionState="entered"
      >
        <p>"The current toasters is of type:warning"</p>
      </DefaultToast>
    );

    expect(container).toMatchSnapshot();
  });
});
