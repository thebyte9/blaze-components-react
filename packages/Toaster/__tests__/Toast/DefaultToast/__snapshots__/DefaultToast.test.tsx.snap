// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default toast should render without throwing error 1`] = `
<div>
  <div
    style="height: 8px; transition: height 120ms 100ms;"
  >
    <div
      class="react-toast-notifications__toast react-toast-notifications__toast--warning"
      style="transition: transform 220ms cubic-bezier(0.2, 0, 0, 1), opacity 220ms; transform: translate3d(0,0,0);"
    >
      <div
        class="react-toast-notifications__toast__icon-wrapper react-toast-notifications__toast__icon-wrapper--warning"
      >
        <div
          class="react-toast-notifications__toast__countdown_animation"
          style="animation-duration: 5000ms; animation-play-state: running; opacity: 1;"
        />
        <svg
          aria-hidden="true"
          class="react-toast-notifications__toast__icon"
          fill="#FFFFFF"
          height="16"
          style="display: inline-block; vertical-align: text-top;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="react-toast-notifications__toast__content"
      >
        <p>
          "The current toasters is of type:warning"
        </p>
      </div>
      <div
        class="react-toast-notifications__toast__dismiss-button"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="react-toast-notifications__toast__dismiss-icon react-toast-notifications__toast__dismiss-icon--warning"
          fill="currentColor"
          height="16"
          style="display: inline-block; vertical-align: text-top;"
          viewBox="0 0 14 16"
          width="14"
        >
          <path
            d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="react-toast-notifications__toast__dismiss-text"
        />
      </div>
    </div>
  </div>
</div>
`;
